src/string_id.h

```diff
@@ -209,11 +214,62 @@ class string_id
         }
 
     private:
-        std::string _id;
+        const char *insert_or_get_cache( const char *string, bool make_unique ) {
+            static std::vector<const char *> sorted_cache;
+            static std::array<const char *, unsorted_cache_size> unsorted_cache;
```

Comment:
I think you can just use `std::set` instead, with `strcmp` passed as the comparator.